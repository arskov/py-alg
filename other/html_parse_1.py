from html.parser import HTMLParser
import sys

class MyHTMLParser(HTMLParser):
    def __init__(self, convert_charrefs=True):
        super(MyHTMLParser, self).__init__(convert_charrefs=True)
        self.result = []

    def handle_starttag(self, tag, attrs):
        self.result.append("Start : {}".format(tag))
        for t in attrs:
            self.result.append("-> {} > {}".format(t[0], t[1]))
    
    def handle_endtag(self, tag):
        self.result.append("End   : {}".format(tag))
    
    def handle_startendtag(self, tag, attrs):
        self.result.append("Empty : {}".format(tag))
        for t in attrs:
            self.result.append("-> {} > {}".format(t[0], t[1]))
    
    def get_result(self):
        return self.result

if __name__ == "__main__":
    parser = MyHTMLParser()
    with open(sys.path[0] + "/html_sample_1.txt") as f, \
        open(sys.path[0] + "/html_parse_1.expected") as fe:
        parser.feed("".join(f.readlines()))
        print(*parser.get_result(), sep = '\n')