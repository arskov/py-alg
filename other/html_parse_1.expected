Start : article
-> class > hentry
Start : div
-> class > entry-content
Start : p
End   : p
Start : p
End   : p
Start : figure
Empty : img
-> src > tammy-strobel.jpg
-> alt > Portrait of Tammy Strobel
Start : figcaption
End   : figcaption
End   : figure
Start : p
End   : p
Start : p
End   : p
Start : aside
Start : h2
End   : h2
Start : ul
Start : li
End   : li
Start : li
End   : li
Start : li
End   : li
End   : ul
End   : aside
Start : div
-> class > entry-content-asset
Start : a
-> href > photo-full.png
Empty : img
-> src > photo.png
-> alt > The objects Tammy removed from her life after moving
End   : a
End   : div
Start : p
End   : p
Start : p
End   : p
Start : a
-> class > entry-unrelated
-> href > http://fake.site/
End   : a
End   : div
Start : footer
Start : p
End   : p
Start : div
-> class > source-org vcard copyright
Start : span
-> class > org fn
End   : span
End   : div
End   : footer
End   : article